golang/internal/bashparser/bashparser.go

```diff
@@ -22,10 +22,10 @@ type CMDResult struct {
 // It returns the parsed commands in JSON format.
 func ParseCommands(data string) (string, error) {
 	// Remove GitHub Actions's expressions because the bash parser doesn't recognize it.
-	// We use greedy matching, so if we have `${{ $ {{ foo }} }}`, it will be matched
-	// to `$MACARON_UNKNOWN`, even though it's not a valid GitHub expression.
+	// We don't use greedy matching, so if we have `${{ $ {{ foo }} }}`, it will not be matched
+	// to `$MACARON_UNKNOWN`.
```

Comment:
You might still want to mention why non-greedy matching was used, particularly since prior use of greedy led to a bug, e.g.
```
// We need to use non-greedy matching, otherwise multiple interpolations e.g. `${{ foo }} bar ${{ baz }}` would be matched incorrectly.
```

Reply:
See commit 10eeb5bcbc5271a479edc50b1c44258e3deccb45